src/main/java/devdojo/exam/transactions/config/SecurityConfig.java

```diff
@@ -1,37 +1,69 @@
 package devdojo.exam.transactions.config;
 
-import devdojo.exam.transactions.repository.BankUserRepository;
-import devdojo.exam.transactions.service.CustomUserDetailsService;
+import devdojo.exam.transactions.domain.BankUser;
 import lombok.RequiredArgsConstructor;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.security.config.Customizer;
 import org.springframework.security.config.annotation.method.configuration.EnableMethodSecurity;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.provisioning.InMemoryUserDetailsManager;
 import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.servlet.util.matcher.MvcRequestMatcher;
+import org.springframework.web.servlet.handler.HandlerMappingIntrospector;
 
 @Configuration
 @EnableMethodSecurity
 @RequiredArgsConstructor
 public class SecurityConfig {
-
-    private final BankUserRepository bankUserRepository;
+    @Bean
```

Comment:
Eu não acho que você precisa desse Bean aqui. Qual o propósito dele?

Reply:
Sem o MvcRequestMatcher.Builder, fico com esse erro ao iniciar a aplicação, 
Caused by: java.lang.IllegalArgumentException: This method cannot decide whether these patterns are Spring MVC patterns or not. If this endpoint is a Spring MVC endpoint, please use requestMatchers(MvcRequestMatcher).
E essa foi a forma que achei de resolver